src/servers/src/proto.rs

```diff
@@ -178,13 +219,8 @@ impl PromWriteRequest {
         self.table_data.as_insert_requests()
     }
 
-    pub fn merge<B>(&mut self, mut buf: B) -> Result<(), DecodeError>
-    where
-        B: Buf,
-        Self: Sized,
-    {
+    pub fn merge(&mut self, mut buf: Bytes) -> Result<(), DecodeError> {
```

Comment:
non-blocking: Maybe it is possible to take `&[u8]` as input and replace `Bytes` with `&[u8]` in these structs. Then the compiler can check the lifetime for us without using unsafe `merge_bytes()` and `split_to`. We can leave a TODO.